lib/ContextHandler.js

```diff
@@ -0,0 +1,44 @@
+"use strict";
+
+module.exports = class ContextHandler {
+    constructor(contextClient) {
+        this.client = contextClient;
+    }
+
+    saveContext(contextId, contextContext) {
+        return new Promise ((resolve, reject) => {
+            this.client.saveContext(contextId, contextContext).then(
+              () => { resolve(); },
+              (error) => { reject(error); }
+            );
+          });  
```

Comment:
Couldn't you just return the promise from saveContext here?
```javascript
return this.client.saveContext(contextId, contextContext);
```

BTW, I'd change ```contextContext``` to ```context```.

Reply:
This also happens in other functions in this file.